src/main/scala/com/databricks/labs/deltaoms/common/OMSSparkConf.scala

```diff
@@ -22,27 +22,82 @@ trait OMSSparkConf extends Serializable with SparkSettings {
 
   def buildConfKey(key: String): String = s"databricks.labs.deltaoms.${key}"
 
+  // Mandatory Configurations
   val BASE_LOCATION = buildConfKey("base.location")
   val DB_NAME = buildConfKey("db.name")
   val CHECKPOINT_BASE = buildConfKey("checkpoint.base")
   val CHECKPOINT_SUFFIX = buildConfKey("checkpoint.suffix")
 
+  // Optional Configuration
+  val RAW_ACTION_TABLE = buildConfKey("raw.action.table")
+  val SOURCE_CONFIG_TABLE = buildConfKey("source.config.table")
+  val PATH_CONFIG_TABLE = buildConfKey("path.config.table")
+  val PROCESSED_HISTORY_TABLE = buildConfKey("processed.history.table")
+  val COMMITINFO_SNAPSHOT_TABLE = buildConfKey("commitinfo.snapshot.table")
+  val ACTION_SNAPSHOT_TABLE = buildConfKey("action.snapshot.table")
+  val CONSOLIDATE_WILDCARD_PATHS = buildConfKey("consolidate.wildcard.paths")
+  val TRUNCATE_PATH_CONFIG = buildConfKey("truncate.path.config")
+  val SKIP_PATH_CONFIG = buildConfKey("skip.path.config")
+  val SKIP_INITIALIZE = buildConfKey("skip.initialize")
+  val SRC_DATABASES = buildConfKey("src.databases")
+  val TABLE_PATTERN = buildConfKey("table.pattern")
+  val TRIGGER_INTERVAL = buildConfKey("trigger.interval")
+  val STARTING_STREAM = buildConfKey("starting.stream")
+  val ENDING_STREAM = buildConfKey("ending.stream")
+
+
+  val configFields = Seq(BASE_LOCATION, DB_NAME, CHECKPOINT_BASE, CHECKPOINT_SUFFIX,
+    RAW_ACTION_TABLE, SOURCE_CONFIG_TABLE, PATH_CONFIG_TABLE, PROCESSED_HISTORY_TABLE,
+    COMMITINFO_SNAPSHOT_TABLE, ACTION_SNAPSHOT_TABLE, CONSOLIDATE_WILDCARD_PATHS,
+    TRUNCATE_PATH_CONFIG, SKIP_PATH_CONFIG, SKIP_INITIALIZE, SRC_DATABASES,
+    TABLE_PATTERN, TRIGGER_INTERVAL, STARTING_STREAM, ENDING_STREAM)
+
   def consolidateOMSConfigFromSparkConf(config: OMSConfig): OMSConfig = {
-    Seq(BASE_LOCATION, DB_NAME, CHECKPOINT_BASE, CHECKPOINT_SUFFIX).foldLeft(config) {
+    configFields.foldLeft(config) {
```

Comment:
You could simplify this a bit, with an internal mapping case class between spark conf and oms conf and also establish required fields that way. something like this. You can also just avoid this internal mapping to maintain by refactoring your original case class `OMSConfig` to use this.

```
/*
 * Copyright (2021) Databricks, Inc.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE
 * AND NONINFRINGEMENT.
 *
 * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
 * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR
 * THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 *
 * See the Full License for the specific language governing permissions and
 * limitations under the License.
 */

package com.databricks.labs.deltaoms.common

import com.databricks.labs.deltaoms.common.OMSSparkConfUtils.{buildConfKey, getSparkConf}
import com.databricks.labs.deltaoms.configuration.{OMSConfig, SparkSettings}

private object OMSSparkConfUtils extends SparkSettings {
  def buildConfKey(key: String): String = s"databricks.labs.deltaoms.${key}"

  def getSparkConf[T](confKey: String): Option[String] = {
    spark.conf.getOption(confKey)
  }
}

case class WithSparkConf[T](value: T,
                            sparkConfigName: String,
                            is_required: Boolean = false)


case class OMSSparkConfig(baseLocation: WithSparkConf[Option[String]] =
                            WithSparkConf(None,
                              buildConfKey("base.location"),
                              is_required = true),
                          dbName: WithSparkConf[Option[String]] =
                          WithSparkConf(None,
                            buildConfKey("db.name"),
                            is_required = true),
                          checkpointBase: WithSparkConf[Option[String]] =
                            WithSparkConf(None,
                              buildConfKey("checkpoint.base"),
                              is_required = true),
                          checkpointSuffix: WithSparkConf[Option[String]] =
                            WithSparkConf(None,
                              buildConfKey("checkpoint.suffix"),
                              is_required = true),
                          rawActionTable: WithSparkConf[String] =
                            WithSparkConf("rawactions",
                              buildConfKey("raw.action.table")),
                          sourceConfigTable: WithSparkConf[String] =
                            WithSparkConf("sourceconfig",
                              buildConfKey("source.config.table")),
                          pathConfigTable: WithSparkConf[String] =
                            WithSparkConf("pathconfig",
                              buildConfKey("path.config.table")),
                          processedHistoryTable: WithSparkConf[String] =
                            WithSparkConf("processedhistory",
                              buildConfKey("processed.history.table")),
                          commitInfoSnapshotTable: WithSparkConf[String] =
                            WithSparkConf("commitinfosnapshots",
                              buildConfKey("commitinfo.snapshot.table")),
                          actionSnapshotTable: WithSparkConf[String] =
                            WithSparkConf("actionsnapshots",
                              buildConfKey("action.snapshot.table")),
                          consolidateWildcardPaths: WithSparkConf[Boolean] =
                            WithSparkConf(true,
                              buildConfKey("consolidate.wildcard.paths")),
                          truncatePathConfig: WithSparkConf[Boolean] =
                            WithSparkConf(false,
                              buildConfKey("truncate.path.config")),
                          skipPathConfig: WithSparkConf[Boolean] =
                            WithSparkConf(false,
                              buildConfKey("skip.path.config")),
                          skipInitializeOMS: WithSparkConf[Boolean] =
                            WithSparkConf(false,
                              buildConfKey("skip.initialize")),
                          srcDatabases: WithSparkConf[Option[String]] =
                            WithSparkConf(None,
                              buildConfKey("src.databases")),
                          tablePattern: WithSparkConf[Option[String]] =
                            WithSparkConf(None,
                              buildConfKey("table.pattern")),
                          triggerInterval: WithSparkConf[Option[String]] =
                            WithSparkConf(None,
                              buildConfKey("trigger.interval")),
                          startingStream: WithSparkConf[Int] =
                            WithSparkConf(1,
                              buildConfKey("starting.stream")),
                          endingStream: WithSparkConf[Int] =
                            WithSparkConf(50,
                              buildConfKey("ending.stream")))

trait OMSSparkConf extends Serializable with SparkSettings {

  def consolidateOMSConfigFromSparkConf(config: OMSConfig): OMSConfig = {
    val sparkOmsConfMap = OMSSparkConfig()
    OMSConfig(
      baseLocation = getSparkConf(sparkOmsConfMap.baseLocation.sparkConfigName)
        .fold(config.baseLocation){Some(_)},
      dbName = getSparkConf(sparkOmsConfMap.dbName.sparkConfigName)
        .fold(config.dbName){Some(_)},
      checkpointBase = getSparkConf(sparkOmsConfMap.checkpointBase.sparkConfigName)
        .fold(config.checkpointBase){Some(_)},
      checkpointSuffix = getSparkConf(sparkOmsConfMap.checkpointSuffix.sparkConfigName)
        .fold(config.checkpointSuffix){Some(_)},
      rawActionTable = getSparkConf(sparkOmsConfMap.rawActionTable.sparkConfigName)
        .fold(config.rawActionTable){_},
      sourceConfigTable = getSparkConf(sparkOmsConfMap.sourceConfigTable.sparkConfigName)
        .fold(config.sourceConfigTable){_},
      pathConfigTable = getSparkConf(sparkOmsConfMap.pathConfigTable.sparkConfigName)
        .fold(config.pathConfigTable){_},
      processedHistoryTable = getSparkConf(sparkOmsConfMap.processedHistoryTable.sparkConfigName)
        .fold(config.processedHistoryTable){_},
      commitInfoSnapshotTable = getSparkConf(sparkOmsConfMap.commitInfoSnapshotTable.sparkConfigName)
        .fold(config.commitInfoSnapshotTable){_},
      actionSnapshotTable = getSparkConf(sparkOmsConfMap.actionSnapshotTable.sparkConfigName)
        .fold(config.actionSnapshotTable){_},
      consolidateWildcardPaths = getSparkConf(sparkOmsConfMap.consolidateWildcardPaths.sparkConfigName)
        .fold(config.consolidateWildcardPaths){_.toBoolean},
      truncatePathConfig = getSparkConf(sparkOmsConfMap.truncatePathConfig.sparkConfigName)
        .fold(config.truncatePathConfig){_.toBoolean},
      skipPathConfig = getSparkConf(sparkOmsConfMap.skipPathConfig.sparkConfigName)
        .fold(config.skipPathConfig){_.toBoolean},
      skipInitializeOMS = getSparkConf(sparkOmsConfMap.skipInitializeOMS.sparkConfigName)
        .fold(config.skipInitializeOMS){_.toBoolean},
      srcDatabases = getSparkConf(sparkOmsConfMap.srcDatabases.sparkConfigName)
        .fold(config.srcDatabases){Some(_)},
      tablePattern = getSparkConf(sparkOmsConfMap.tablePattern.sparkConfigName)
        .fold(config.tablePattern){Some(_)},
      triggerInterval = getSparkConf(sparkOmsConfMap.triggerInterval.sparkConfigName)
        .fold(config.triggerInterval){Some(_)},
      startingStream = getSparkConf(sparkOmsConfMap.startingStream.sparkConfigName)
        .fold(config.startingStream){_.toInt},
      endingStream = getSparkConf(sparkOmsConfMap.endingStream.sparkConfigName)
        .fold(config.endingStream){_.toInt}
  }
}

object OMSSparkConf extends OMSSparkConf
```

Reply:
Thanks @bali0019 . Will definitely look into this. Marking as TODO for now. Opened the issue : https://github.com/databrickslabs/delta-oms/issues/19